report/simple_report.go

```diff
@@ -0,0 +1,174 @@
+package report
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const SEV_LOW = "low"
+const SEV_MEDIUM = "medium"
+const SEV_HIGH = "high"
+const SEV_CRITICAL = "critical"
+
+type Risk struct {
+	Severity    string   `json:"severity"`
+	Title       string   `json:"title"`
+	Description string   `json:"description"`
+	Fix         string   `json:"fix"`
+	Items       []string `json:"items"`
+}
+
+type SimpleReport struct {
+	HiddenService string `json:"hiddenService"`
+	Risks         []Risk `json:"risks"`
+}
+
+func (osr *SimpleReport) AddRisk(severity string, title string, description string, fix string, items []string) {
+	osr.Risks = append(osr.Risks, Risk{severity, title, description, fix, items})
+}
+
+// Format as JSON
+func (osr *SimpleReport) Serialize() (string, error) {
+	report, err := json.Marshal(osr)
+	if err != nil {
+		return "", err
+	}
+	return string(report), nil
+}
+
+var risk_levels = map[string]string{
+	SEV_LOW:      "\033[094mLow Risk:\033[0m",
+	SEV_MEDIUM:   "\033[093mMedium Risk:\033[0m",
+	SEV_HIGH:     "\033[091mHigh Risk:\033[0m",
+	SEV_CRITICAL: "\033[091mCritical Risk:\033[0m",
+}
+
+// Format as human-readable text to be printed to console
+func (osr *SimpleReport) Format() (string, error) {
+	buffer := bytes.NewBuffer(nil)
+	buffer.WriteString("--------------- OnionScan Report ---------------\n")
+
+	buffer.WriteString(fmt.Sprintf("Generating Report for: %s\n\n", osr.HiddenService))
+
+	for _, risk := range osr.Risks {
+		buffer.WriteString(risk_levels[risk.Severity] + " " + risk.Title + "\n")
+		if len(risk.Description) > 0 {
+			for _, item := range strings.Split(risk.Description, "\n") {
+				buffer.WriteString("\t " + item + "\n")
+			}
+		}
+		if len(risk.Fix) > 0 {
+			for _, item := range strings.Split(risk.Fix, "\n") {
+				buffer.WriteString("\t " + item + "\n")
+			}
+		}
+		if len(risk.Items) > 0 {
+			buffer.WriteString("\t Items Identified:\n")
+			buffer.WriteString("\n")
+			for _, item := range risk.Items {
+				buffer.WriteString(fmt.Sprintf("\t %s\n", item))
+			}
+		}
+		buffer.WriteString("\n")
+
+	}
+	return buffer.String(), nil
+}
+
+func SummarizeToSimpleReport(report *AnonymityReport) *SimpleReport {
+	var out = NewSimpleReport(report.OnionScanReport.HiddenService)
+
+	if len(report.EmailAddresses) > 0 {
+		out.AddRisk(SEV_LOW, "Found Identities", "", "", report.EmailAddresses)
+	}
+
+	if len(report.IPAddresses) > 0 {
+		out.AddRisk(SEV_LOW, "Found IP Addresses", "", "", report.IPAddresses)
+	}
+
+	if len(report.AnalyticsIDs) > 0 {
+		out.AddRisk(SEV_LOW, "Found Analytics IDs", "", "", report.AnalyticsIDs)
+	}
+
+	if len(report.BitcoinAddresses) > 0 {
+		out.AddRisk(SEV_LOW, "Found Bitcoin Addresses", "", "", report.BitcoinAddresses)
+	}
+
+	if report.FoundApacheModStatus {
+		out.AddRisk(SEV_HIGH, "Apache mod_status is enabled and accessible",
```

Comment:
One thing I could do is to have these as a list of abstract "SimpleReportRule"s which can be plugged in instead of lumping them together in a function.